easybuild/easyblocks/l/lammps.py

```diff
@@ -118,6 +118,8 @@
     'zen2': 'ZEN2',
     'zen3': 'ZEN3',
     'power9le': 'POWER9',
+    'neoverse_n1': 'ARMV81',
+    'neoverse_v1': 'ARMV81',
```

Comment:
Recent LAMMPS mapping is at https://docs.lammps.org/Build_extras.html#available-architecture-settings

Reply:
We are in a tricky situation here as you also need to update the `KOKKOS_CPU_MAPPING ` and `KOKKOS_GPU_ARCH_TABLE` to match up the new architectures introduced _but_ this has to be done in a backwards compatible way. Probably we need to allow that the values of these dictionaries may be dictionaries themselves which  version keys and the appropriate return values.

Reply:
Should I than test this easyblock with older easyconfigs of LAMMPS to see if anything is broken?

Reply:
Or update the "constants" based on the LAMMPS version in the easyblock constructor?

Maybe using constants was just wrong here, and it needs to be done via class variables like `self.kokkos_cpu_mapping` instead?

Reply:
We could start doing something like this to update the mapping for particular versions of LAMMPS:

```python
    def update_kokkos_cpu_mapping(self):
                
        if LooseVersion(self.cur_version) >= LooseVersion(translate_lammps_version('2Aug2023')):
            self.kokkos_cpu_mapping['a64fx'] = 'A64FX'
    
    def __init__(self, *args, **kwargs):
        """LAMMPS easyblock constructor: determine whether we should build with CUDA support enabled."""
        super(EB_LAMMPS, self).__init__(*args, **kwargs)

        self.kokkos_cpu_mapping = copy.deepcopy(KOKKOS_CPU_MAPPING)
        self.update_kokkos_cpu_mapping()
```

and keep `KOKKOS_CPU_MAPPING` as is, assuming that it's correct for all LAMMPS versions we care about

`ARMV81` is supported since `stable_29Oct2020` it seems, see https://github.com/lammps/lammps/commit/60864e38d1d5f34db0ff379bf71eab0da0ca1ed0

Reply:
`ARMv80`, `ARMv81`, `ARMv8-ThunderX` are actually supported since `stable_31Mar2017`, see https://github.com/lammps/lammps/commit/a9f0b7d523fa86538d1b1b19e3da8a049ad3f439

Reply:
`ARMv8-TX2` is available since `stable_29sep2021`  (see https://github.com/lammps/lammps/commit/39786b17407dd03a501f8582df53bd56a25d16f7) and `A64FX` is available since 'stable_29Sep2021` (see https://github.com/lammps/lammps/commit/eea14c55a9bbbe33423f2160f6949fc41ecb6a95)